### PR TITLE
test: add integration testing for az iot ops init

### PR DESCRIPTION
**Pull request by vilit1**
image

image

image

image

latest runs:
--simulate-plc
https://github.com/Azure/azure-iot-ops-cli-extension/actions/runs/10151869019

no input:
https://github.com/Azure/azure-iot-ops-cli-extension/actions/runs/10151870322

https://github.com/Azure/azure-iot-ops-cli-extension/pull/254
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

File: dataflow_crd.yaml
Comment (digimaun) [inline dataflow_crd.yaml:1]: Why are there dataflow CRDs in this PR ?

Comment (vilit1) [reply dataflow_crd.yaml:1]: idk

File: .github/workflows/int_test.yml [hunk 1]
```diff
@@ -84,35 +95,49 @@ jobs:
...
     runs-on: ubuntu-22.04
     steps:
+      - name: "Determine Init Args"
+        id: "init"
+        run: |
+          echo "NO_PREFLIGHT=false" >> $GITHUB_OUTPUT
+          if [[ ${{ matrix.feature }} == "default" ]]; then
+            echo "ARG=\"--ca-valid-days 3 --kv-spc-secret-name test-kv-secret\"" >> $GITHUB_OUTPUT
+          elif [[ ${{ matrix.feature }} == "mq-insecure" ]]; then
+            echo "ARG=\"--add-insecure-listener --csi-config telegraf.resources.limits.memory=500Mi telegraf.resources.limits.cpu=100m\"" >> $GITHUB_OUTPUT
```
Comment (vilit1) [inline .github/workflows/int_test.yml:127]: add in `--broker-service-type LoadBalancer`

Comment (digimaun) [inline .github/workflows/int_test.yml:127]: Please include a test run that has `--broker-service-type LoadBalancer` that can be combined with `--add-insecure-listener`.

File: .github/workflows/int_test.yml [hunk 2]
```diff
@@ -84,35 +95,49 @@ jobs:
...
       K3S_VERSION: "v1.28.5+k3s1"
+      CA_FILE: "test-ca.pem"
+      CA_KEY_FILE: "test-ca-key.pem"
     strategy:
       fail-fast: false
       matrix:
-        feature: [default, mq-insecure, no-syncrules, ca-certs]
-        include:
-          # default / limited options
+        feature: [custom-input, default, mq-insecure, no-syncrules, ca-certs]
```
Comment (digimaun) [inline .github/workflows/int_test.yml:103]: mq-insecure should probably be renamed to "insecure-listener" or similar.

File: .github/workflows/int_test.yml [hunk 3]
```diff
@@ -84,35 +95,49 @@ jobs:
...
-            ca-file: 'test-ca.pem'
-            ca-key-file: 'test-ca-key.pem'
+            runtime-args: true
     name: "Run cluster tests"
     runs-on: ubuntu-22.04
     steps:
+      - name: "Determine Init Args"
+        id: "init"
+        run: |
+          echo "NO_PREFLIGHT=false" >> $GITHUB_OUTPUT
```
Comment (digimaun) [inline .github/workflows/int_test.yml:123]: `--simulate-plc` should be ran and evaluated

File: .github/workflows/int_test.yml [hunk 4]
```diff
@@ -84,35 +95,49 @@ jobs:
...
+          if [[ ${{ matrix.feature }} == "default" ]]; then
+            echo "ARG=\"--ca-valid-days 3 --kv-spc-secret-name test-kv-secret\"" >> $GITHUB_OUTPUT
+          elif [[ ${{ matrix.feature }} == "mq-insecure" ]]; then
+            echo "ARG=\"--add-insecure-listener --csi-config telegraf.resources.limits.memory=500Mi telegraf.resources.limits.cpu=100m\"" >> $GITHUB_OUTPUT
+            echo "NO_PREFLIGHT=true" >> $GITHUB_OUTPUT
+          elif [[ ${{ matrix.feature }} == "default" ]]; then
```
Comment (vilit1) [inline .github/workflows/int_test.yml:129]: why is this default?

**Review by digimaun — APPROVED** [review]

File: .github/workflows/int_test.yml [hunk 5]
```diff
@@ -297,33 +311,43 @@ jobs:
           az iot ops broker stats
           az iot ops broker stats --raw
           az iot ops asset query -g ${{ env.RESOURCE_GROUP }} --location westus -o table
-          az iot ops verify-host
+          # az iot ops verify-host
```
Comment (digimaun) [inline .github/workflows/int_test.yml:314]: What was the reason?

Comment (vilit1) [reply .github/workflows/int_test.yml:314]: probably leftover from commenting out broken stuff...

**Pull request merged**